src/mapping/device/GinkgoRBFKernels.cpp

```diff
@@ -1,132 +1,188 @@
+std::shared_ptr<gko::Executor> create_device_executor(const std::string &execName, bool enableUnifiedMemory)
+{
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactPolynomialC0>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                      double *, double *, double *, precice::mapping::CompactPolynomialC0, const RadialBasisParameters,
-                                                                                      const std::size_t, const std::size_t, const bool, const unsigned int);
+#ifdef KOKKOS_ENABLE_SERIAL
+  if (execName == "reference-executor") {
+    return gko::ext::kokkos::create_executor(Kokkos::Serial{});
+  }
+#endif
+  if (enableUnifiedMemory) {
+    return gko::ext::kokkos::create_executor(Kokkos::DefaultExecutionSpace{},
+                                             Kokkos::SharedSpace{});
+  } else {
+    return gko::ext::kokkos::create_executor(Kokkos::DefaultExecutionSpace{});
+  }
+}
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactPolynomialC2>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                      double *, double *, double *, precice::mapping::CompactPolynomialC2, const RadialBasisParameters,
-                                                                                      const std::size_t, const std::size_t, const bool, const unsigned int);
+namespace kernel {
+
+template <typename MemorySpace, typename EvalFunctionType>
+void create_rbf_system_matrix_impl(std::shared_ptr<const gko::Executor>      exec,
+                                   gko::ptr_param<GinkgoMatrix>              mtx,
+                                   const std::array<bool, 3>                 activeAxis,
+                                   gko::ptr_param<GinkgoMatrix>              supportPoints,
+                                   gko::ptr_param<GinkgoMatrix>              targetPoints,
+                                   EvalFunctionType                          f,
+                                   ::precice::mapping::RadialBasisParameters rbf_params,
+                                   bool                                      addPolynomial,
+                                   unsigned int                              extraDims)
+{
+  auto k_mtx           = map_data<MemorySpace>(mtx.get());
+  auto k_supportPoints = map_data<MemorySpace>(supportPoints.get());
+  auto k_targetPoints  = map_data<MemorySpace>(targetPoints.get());
+
+  // the outcome of this if-condition is known at compile time. However, Cuda
+  // has issues with constexpr
+  if (std::is_same_v<MemorySpace, Kokkos::HostSpace>) {
```

Comment:
Does it make sense to split the function up into two?

The general col-major version and then specialize it for MemorySpace `Kokkos::HostSpace`?

Reply:
In principle, this states an alternative, yes. However, this would require more boilerplate code and I don't see immediate benefit or the upside.

Reply:
Specializing doesn't require much boilerplate. It could actually be net 0 in this case.

Reply:
```bash 
error: non-class, non-variable partial specialization ‘create_rbf_system_matrix_impl<Kokkos::HostSpace, EvalFunctionType>’ is not allowed
   48 | void create_rbf_system_matrix_impl<Kokkos::HostSpace, EvalFunctionType>
```

and the net
 1 file changed, 82 insertions(+), 57 deletions(-)

Repeating the signature is already significant. So I would leave it as is?